source/gui/guiHelper.py

```diff
@@ -140,35 +158,25 @@ def associateElements(firstElement: wx.Control, secondElement: wx.Control) -> wx
 
 	# staticText and input control
 	# likely a labelled control from LabeledControlHelper
-	if isinstance(firstElement, wx.StaticText) and isinstance(
-		secondElement,
-		(
-			wx.Button,
-			wx.Choice,
-			wx.Slider,
-			wx.SpinCtrl,
-			wx.TextCtrl,
-		),
-	):
-		sizer = wx.BoxSizer(wx.HORIZONTAL)
-		sizer.Add(firstElement, flag=wx.ALIGN_CENTER_VERTICAL)
-		sizer.AddSpacer(SPACE_BETWEEN_ASSOCIATED_CONTROL_HORIZONTAL)
-		sizer.Add(secondElement)
-	elif isinstance(firstElement, wx.StaticText) and isinstance(secondElement, wx.CheckBox):
-		# checkbox should go first, and label should go after
-		sizer = wx.BoxSizer(wx.HORIZONTAL)
-		sizer.Add(secondElement)
-		sizer.AddSpacer(SPACE_BETWEEN_ASSOCIATED_CONTROL_HORIZONTAL)
-		sizer.Add(firstElement, flag=wx.ALIGN_CENTER_VERTICAL)
-	# staticText and (ListCtrl, ListBox or TreeCtrl)
-	elif isinstance(firstElement, wx.StaticText) and isinstance(
-		secondElement,
-		(wx.ListCtrl, wx.ListBox, wx.TreeCtrl),
-	):
-		sizer = wx.BoxSizer(wx.VERTICAL)
-		sizer.Add(firstElement)
-		sizer.AddSpacer(SPACE_BETWEEN_ASSOCIATED_CONTROL_VERTICAL)
-		sizer.Add(secondElement, flag=wx.EXPAND, proportion=1)
+	if isinstance(firstElement, wx.StaticText):
+		# Horizontal layout, label first, control second
+		if isinstance(secondElement, _HorizontalCtrlT.__constraints__):
+			sizer = wx.BoxSizer(wx.HORIZONTAL)
+			sizer.Add(firstElement, flag=wx.ALIGN_CENTER_VERTICAL)
+			sizer.AddSpacer(SPACE_BETWEEN_ASSOCIATED_CONTROL_HORIZONTAL)
+			sizer.Add(secondElement)
+		# Horizontal layout, control first, label second
+		elif isinstance(secondElement, wx.CheckBox):
```

Comment:
This code was already present in the old version.
But I wonder why this case has been defined since wx.Checkbox already has its own included label and, visually, the checkbox is already located left and its label right.
Adding any comment to indicate when this case can be useful would be a plus.

Reply:
I'm going to consider this out of scope